src/Main.tsx

```diff
@@ -1,25 +1,41 @@
 import createBrowserHistory from 'history/createBrowserHistory';
 import * as React from 'react';
 import { Redirect, Route, Router, Switch } from 'react-router';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+
 import { setupAppContext, theAppContext } from './AppContext';
 import { BalancesView } from './balances/BalancesView';
 import { ExchangeViewTxRx } from './exchange/ExchangeView';
-import { HeaderWithRouter } from './header/Header';
+import { Header } from './header/Header';
 import * as styles from './index.scss';
 
 const browserHistoryInstance = createBrowserHistory();
 
 export class Main extends React.Component {
   public render() {
-    return (<theAppContext.Provider value={setupAppContext()}>
-      <Router history={browserHistoryInstance}>
+    return (
+      <theAppContext.Provider value={setupAppContext()}>
+        <Router history={browserHistoryInstance}>
+          <MainContentWithRouter />
+        </Router>
+      </theAppContext.Provider>
+    );
+  }
+}
+
+interface RouterProps extends RouteComponentProps<any>{}
```

Comment:
If there are no additional props why to create an "empty" interface? I guess if we need in the future to extend the standard `RouteComponentProps` then we can add that interface

Reply:
The convention is to name the Props type, which can either extend an interface, contain some fields, or both extend and add fields. The app has other instances of Props that extends but adds no fields